### ISSUE TITLE
[rust] move the Rust wrappers into its own separate repository

### ISSUE THREAD

**Issue by joseluis** [dankamongmen/notcurses#2171]
We talked a long time ago about this, and the recent issues related to the changes to the input system reminded me of it again.

I still very much think this would be a change for the best in all regards.

They're really separate projects with separate rhythms of development, and the current system of having both version numbers automatically fully-synced, and both source codes affecting each other on every commit is not worth all the practical disadvantages IMHO.

Firstly, rigth now the version number in the rust wrappers is devoid of meaning and in fact plain misleading. The version is bumped at the same time as the notcurses C library, but the rust wrappers can't ever be really in sync with the features of that version, until they're updated later, condemned to be always behind by design.

Furthermore, a breaking change in either the C library or the Rust bindings can break the other project, and it happens regularly. If they were in separate repositories the Rust wrappers could have feature-complete stable releases against specific version of the notcurses C library, and users could actually depend on them more than is currently possible.

The repository could be in e.g. dankamongmen/libnotcurses-sys.

**Comment by dankamongmen** [comment]
let's do it

**Comment by dankamongmen** [comment]
proceed as you see fit. i'd say just move it on out and try to keep what history you can. i know we discussed this at length before. if you need me to sign you over a repo, or at least make you a full collaborator, just let me know.

**Comment by joseluis** [comment]
Ok thanks if you could just create the repo `dankamongmen/libnotcurses-sys` and add me so that I can start configuring it I can begin setting things up.

**Comment by dankamongmen** [comment]
> Ok thanks if you could just create the repo `dankamongmen/libnotcurses-sys` and add me so that I can start configuring it I can begin setting things up.

it is done

**Comment by joseluis** [comment]
thanks the repo exists but I dont think I have permissions

**Comment by dankamongmen** [comment]
> thanks the repo exists but I dont think I have permissions

i sent you an invite.; it currently says "pending"

**Comment by joseluis** [comment]
> i sent you an invite.; it currently says "pending"

lol, sorry, I forgot the system worked via invitation... Done!

**Comment by joseluis** [comment]
The new repository is up, with the relevant history merged. And I've removed the rust bindings from this repo.

Things that remain:
- [ ] update the readme and/or NEWS.md in this repo.
- [ ] configure the new repo to test the rust bindings in different platforms similarly as before

**Comment by dankamongmen** [comment]
I updated the `README.md` and `NEWS.md`.

**Comment by dankamongmen** [comment]
do you need any help setting up the workflows? i think we can close this bug out from this side. when do you anticipate your first release? i'll want to upload it to debian and fedora.

**Comment by joseluis** [comment]
> do you need any help setting up the workflows?

yes, thanks, I would apreciate that

> i think we can close this bug out from this side.

Agreed. I've opened an issue at https://github.com/dankamongmen/libnotcurses-sys/issues/5

> when do you anticipate your first release? i'll want to upload it to debian and fedora.

I'm not sure, but I'd like the features to be on par for a given release before publishing. I've opened an issue for that at https://github.com/dankamongmen/libnotcurses-sys/issues/6 and I'm gonna start munching on that list of pending things.
